src/engine/Operation.cpp

```diff
@@ -159,8 +159,10 @@ std::shared_ptr<const Result> Operation::getResult(
         result.applyLimitOffset(_limit);
         runtimeInfo().addLimitOffsetRow(_limit, limitTimer.msecs(), true);
       } else {
-        AD_CONTRACT_CHECK(result.idTable().numRows() ==
-                          _limit.actualSize(result.idTable().numRows()));
+        auto numRows = result.idTable().numRows();
```

Comment:
Small nitpick, but I think if `size_t` is used here instead of auto no explicit cast might be necessary